staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -884,6 +881,19 @@ func (r *crdHandler) getOrCreateServingInfoFor(uid types.UID, name string) (*crd
 			SelfLinkPathPrefix: selfLinkPrefix,
 			SelfLinkPathSuffix: "/status",
 		}
+
+		if utilfeature.DefaultFeatureGate.Enabled(features.ServerSideApply) {
```

Comment:
I'm still working on crds. The test still fails for both crds and crs. Not sure why.
The resetFields get added correctly, maybe i messed up something here.
If somebody has time, please take a look. I'll continue here as soon as i can.